src/data/list/zip.lean

```diff
@@ -242,4 +264,37 @@ begin
         simp * at *, }, }, },
 end
 
+@[simp] lemma map_uncurry_zip_eq_zip_with
+  (f : α → β → γ) (l : list α) (l' : list β) :
+  map (function.uncurry f) (l.zip l') = zip_with f l l' :=
+begin
+  induction l with hd tl hl generalizing l',
+  { simp },
+  { cases l' with hd' tl',
+    { simp },
+    { simp [hl] } }
+end
+
+@[simp] lemma sum_zip_with_distrib_left [semiring γ]
+  (f : α → β → γ) (n : γ) (l : list α) (l' : list β) :
+  (zip_with (λ x y, n * f x y) l l').sum = n * (l.zip_with f l').sum :=
```

Comment:
This lemma seems overly specific to me, so I'd be happier if you removed it from this PR (and I'm fairly certain it shouldn't be a `simp` lemma since it is so specialised). If it stays, I would prove it by noting that `zip_with` is a `map`, per the previous lemma, and that `sum (map (λ x, n * x) l) = n * sum l`. If you remove it, you can inline this proof at the use site.

Reply:
Removed, I'll inline it at the use site in the relevant PR